lib/matplotlib/testing/__init__.pyi

```diff
@@ -0,0 +1,48 @@
+from collections.abc import Callable
+import subprocess
+from typing import Any, IO, Literal, overload
+
+def set_font_settings_for_testing() -> None: ...
+def set_reproducibility_for_testing() -> None: ...
+def setup() -> None: ...
+@overload
+def subprocess_run_for_testing(
+    command: list[str],
+    text: Literal[True],
+    env: dict[str, str] | None = ...,
+    timeout: float | None = ...,
+    stdout: int | IO[Any] | None = ...,
+    stderr: int | IO[Any] | None = ...,
+    check: bool = ...,
+    capture_output: bool = ...,
+) -> subprocess.CompletedProcess[str]: ...
+@overload
+def subprocess_run_for_testing(
+    command: list[str],
+    text: Literal[False],
+    env: dict[str, str] | None = ...,
+    timeout: float | None = ...,
+    stdout: int | IO[Any] | None = ...,
+    stderr: int | IO[Any] | None = ...,
+    check: bool = ...,
+    capture_output: bool = ...,
+) -> subprocess.CompletedProcess[bytes]: ...
+@overload
+def subprocess_run_for_testing(
+    command: list[str],
+    env: dict[str, str] | None = ...,
+    timeout: float | None = ...,
+    stdout: int | IO[Any] | None = ...,
+    stderr: int | IO[Any] | None = ...,
+    check: bool = ...,
+    text: bool = ...,
+    capture_output: bool = ...,
+) -> subprocess.CompletedProcess[bytes] | subprocess.CompletedProcess[str]: ...
```

Comment:
I think I see what you are going for/why you did it, but the reordering of the `text` argument is not actually consistent with the runtime signature. (It presents itself as a positional or keyword arg in the first two signatures, but it doesn't go there positionally.

I think the three overloads are:

- `text=True` explicitly passed as a required keyword (at which point not having a default is allowed)
- `text=True` is passed positionally (probably rare with this many arguments, but allowed), at which point _every_ previous argument must also be passed positionally, and therefore not have a default, making it allowed
- `text=False` is passed, or used as the default, positionally or keyword.

You may still wish to keep the fallback where text is a `bool` and you get a union output, but that is really only for the case where someone is passing a bool variable around, which seems rare (and even if it is a local var, the type checker will thread through the literal, so it needs to be doing some comparison or function call to get generic bool)

```suggestion
@overload
def subprocess_run_for_testing(
    command: list[str],
    env: dict[str, str] | None = ...,
    timeout: float | None = ...,
    stdout: int | IO[Any] | None = ...,
    stderr: int | IO[Any] | None = ...,
    check: bool = ...,
    *,
    text: Literal[True],
    capture_output: bool = ...,
) -> subprocess.CompletedProcess[str]: ...
@overload
def subprocess_run_for_testing(
    command: list[str],
    env: dict[str, str] | None,
    timeout: float | None,
    stdout: int | IO[Any] | None,
    stderr: int | IO[Any] | None,
    check: bool,
    text: Literal[True],
    capture_output: bool = ...,
) -> subprocess.CompletedProcess[str]: ...
@overload
def subprocess_run_for_testing(
    command: list[str],
    env: dict[str, str] | None = ...,
    timeout: float | None = ...,
    stdout: int | IO[Any] | None = ...,
    stderr: int | IO[Any] | None = ...,
    check: bool = ...,
    text: Literal[False] = ...,
    capture_output: bool = ...,
) -> subprocess.CompletedProcess[bytes]: ...
```

Reply:
Ah, it didn't seem to like what I tried, but I think the `*,` might be the key here that I didn't try.